src/qibolab/platform.py

```diff
@@ -20,6 +21,34 @@
 QubitPairMap = Dict[QubitPairId, QubitPair]
 
 
+def unroll_sequences(sequences: List[PulseSequence], relaxation_time: int) -> PulseSequence:
+    """Unrolls a list of pulse sequences to a single pulse sequence with multiple measurements.
+
+    Args:
+        sequences (list): List of pulse sequences to unroll.
+        relaxation_time (int): Time in ns to wait for the qubit to relax between
+            playing different sequences.
+
+    Returns:
+        total_sequence (:class:`qibolab.pulses.PulseSequence`): Unrolled pulse sequence containing
+            multiple measurements.
+        readout_map (dict): Map from original readout pulse serials to the unrolled readout pulse
+            serials. Required to construct the results dictionary that is returned after execution.
+    """
+    total_sequence = PulseSequence()
+    readout_map = defaultdict(list)
+    start = 0
+    for sequence in sequences:
+        for pulse in sequence:
+            new_pulse = pulse.copy()
+            new_pulse.start += start
+            total_sequence.add(new_pulse)
+            if isinstance(pulse, ReadoutPulse):
+                readout_map[pulse.serial].append(new_pulse.serial)
+        start = total_sequence.finish + relaxation_time
+    return total_sequence, readout_map
```

Comment:
I am not sure if that's true. We would still need to unroll the list of sequences to a single sequence with proper `WaitPulse`s between the sequences. It would indeed be a bit simpler though as we won't need to shift the start of every pulse.

Reply:
This is something we could with a suitable iteration, possibly using `itertools` (e.g. `concatenate`).

Reply:
#682